### PR TITLE
octoprint: add nixosTests

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

This adds some basic tests of the `octoprint` service. These will hopefully catch regressions like #213145.

Please note: Due to the PR above not yet being merged, this nixosTest will fail (as it should), but will pass, once the commit has been applied.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
Please rebase.

File: pkgs/applications/misc/octoprint/default.nix [hunk 1]
```diff
@@ -202,6 +203,7 @@ let
               passthru = {
                 python = self.python;
                 updateScript = nix-update-script { };
+                tests = nixosTests.octoprint;
```
Comment (NickCao) [inline pkgs/applications/misc/octoprint/default.nix:206]: passthru.tests is expected to be an attrset.
```suggestion
                tests = {
                  inherit (nixosTests) octoprint;
                };
```

File: nixos/tests/octoprint.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }:
+
+let
+  apikey = "testapikey";
+in
+{
+  name = "paperless";
```
Comment (NickCao) [inline nixos/tests/octoprint.nix:7]: Leftover from copy pasting?

Comment (gador) [reply nixos/tests/octoprint.nix]: yes..

**Comment by gador** [comment]
> Please rebase.

done :+1:

**Comment by NickCao** [comment]
@ofborg test octoprint

**Review by NickCao — APPROVED** [review]

**Pull request merged**